### ISSUE TITLE
Make JdtBasedProcessorProvider work non-default eclipse file systems

### ISSUE THREAD

**Issue by cdietrich** [eclipse/xtext-xtend#50]
JdtBasedProcessorProvider crashes when class path elements (libary jars) are not backed by the default eclipse file system but another impl e.g. IBM Jazz.

```
java.net.MalformedURLException: unknown protocol: sourcecontrol
       at java.net.URL.<init>(Unknown Source)
       at java.net.URL.<init>(Unknown Source)
       at java.net.URL.<init>(Unknown Source)
       at java.net.URI.toURL(Unknown Source)
       at org.eclipse.xtend.ide.macro.JdtBasedProcessorProvider.collectClasspathURLs(JdtBasedProcessorProvider.java:239)
       at org.eclipse.xtend.ide.macro.JdtBasedProcessorProvider.createClassLoaderForJavaProject(JdtBasedProcessorProvider.java:142)
       at org.eclipse.xtend.ide.macro.JdtBasedProcessorProvider.getClassLoader(JdtBasedProcessorProvider.java:111)
       at org.eclipse.xtend.core.macro.ConstantExpressionsInterpreter.evaluate(ConstantExpressionsInterpreter.java:180)
       at org.eclipse.xtend.core.macro.XAnnotationExtensions.getProcessorType(XAnnotationExtensions.java:245)
       at org.eclipse.xtend.core.macro.ActiveAnnotationContextProvider$1.accept(ActiveAnnotationContextProvider.java:94)
       at org.eclipse.xtend.core.macro.ActiveAnnotationContextProvider$1.accept(ActiveAnnotationContextProvider.java:1)
       at org.eclipse.xtend.core.macro.ActiveAnnotationContextProvider.registerMacroAnnotations(ActiveAnnotationContextProvider.java:309)
       at org.eclipse.xtend.core.macro.ActiveAnnotationContextProvider.searchAnnotatedElements(ActiveAnnotationContextProvider.java:203)
       at org.eclipse.xtend.core.macro.ActiveAnnotationContextProvider.access$3(ActiveAnnotationContextProvider.java:185)
       at org.eclipse.xtend.core.macro.ActiveAnnotationContextProvider$2.apply(ActiveAnnotationContextProvider.java:194)
       at org.eclipse.xtend.core.macro.ActiveAnnotationContextProvider$2.apply(ActiveAnnotationContextProvider.java:1)
       at org.eclipse.xtext.xbase.lib.IteratorExtensions.forEach(IteratorExtensions.java:363)
       at org.eclipse.xtext.xbase.lib.IterableExtensions.forEach(IterableExtensions.java:333)
```

the problematic piece of code is

```
library.rawLocationURI.toURL
```

`libary.rawLocationURI` will contain something like `sourcecontrol:/jazz/default/....jar` so that toURL will fail.

we should consider to use something like

```
libary.rawLocation.toFile.toURI.toURL
```

or

```
libary.rawLocation.makeAbsolute.toFile.toURI.toURL
```

or

```
EFS.getStore(libary.rawLocation).toLocalFile(0, null).toURI.toURL
```

(needs to be tested with linked jars as well)

**Comment by cdietrich** [comment]
the very same issue was reported here before: https://www.eclipse.org/forums/index.php/m/1735117/

**Comment by lkolek** [comment]
I have the same problem with kotlin plugin installed:


```
java.net.MalformedURLException: unknown protocol: org.jetbrains.kotlin.core.filesystem
	at java.net.URL.<init>(URL.java:593)
	at java.net.URL.<init>(URL.java:483)
	at java.net.URL.<init>(URL.java:432)
	at java.net.URI.toURL(URI.java:1089)
	at org.eclipse.xtend.ide.macro.JdtBasedProcessorProvider.collectClasspathURLs(JdtBasedProcessorProvider.java:174)
	at org.eclipse.xtend.ide.macro.JdtBasedProcessorProvider.collectClasspathURLs(JdtBasedProcessorProvider.java:167)
	at org.eclipse.xtend.ide.macro.JdtBasedProcessorProvider.createClassLoaderForJavaProject(JdtBasedProcessorProvider.java:115)
	at org.eclipse.xtend.ide.macro.JdtBasedProcessorProvider.getClassLoader(JdtBasedProcessorProvider.java:90)
	at org.eclipse.xtend.core.macro.ConstantExpressionsInterpreter.evaluate(ConstantExpressionsInterpreter.java:158)
	at org.eclipse.xtend.core.macro.XAnnotationExtensions.getProcessorType(XAnnotationExtensions.java:209)
	at org.eclipse.xtend.core.macro.ActiveAnnotationContextProvider.lambda$0(ActiveAnnotationContextProvider.java:86)
	at org.eclipse.xtend.core.macro.ActiveAnnotationContextProvider$$Lambda$279/1368249689.accept(Unknown Source)
	at org.eclipse.xtend.core.macro.ActiveAnnotationContextProvider.registerMacroAnnotations(ActiveAnnotationContextProvider.java:251)
	at org.eclipse.xtend.core.macro.ActiveAnnotationContextProvider.searchAnnotatedElements(ActiveAnnotationContextProvider.java:174)
```

**Comment by cdietrich** [comment]
created https://github.com/eclipse/xtext/issues/2276 to find a general solution

**Comment by cdietrich** [comment]
Fixed in Xtext 2.14
